pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -335,6 +335,13 @@ public synchronized void readMoreEntries() {
             }
             return;
         }
+        if (havePendingReplayRead) {
```

Comment:
I am considering the changes in this class is unnecessary, could you remove them?

Reply:
Ok, I'll move the havePendingReplayRead state handling changes to a separate PR. It's strictly not necessary for addressing the issue that the test reproduced.

Reply:
In any case, I'll move the havePendingReplayRead changes to another PR. Reproducing the race condition referred to in the previous comment is hard to reproduce and most likely reproducing would require introducing delays. The PulsarMockBookKeeper read handle interceptor in #23875 would be useful for that purpose. @poorbarcode Could you please review it in the mean time? It should be a quick review to cover #23875 and #23876.

Reply:
@poorbarcode havePendingReplayRead state handling changes have been removed. PTAL